src/Cli/dotnet/Commands/MSBuild/MSBuildForwardingApp.cs

```diff
@@ -34,6 +39,28 @@ private static MSBuildArgs ConcatTelemetryLogger(MSBuildArgs msbuildArgs)
         return msbuildArgs;
     }
 
+    /// <summary>
+    /// Adjusts MSBuild loggers to be more suitable for LLM/agentic environments, if such an environment is detected.
+    /// </summary>
+    /// <param name="msbuildArgs"></param>
+    /// <returns></returns>
+    private static MSBuildArgs AdjustLoggerForLLMs(MSBuildArgs msbuildArgs)
+    {
+        if (new Telemetry.LLMEnvironmentDetectorForTelemetry().GetLLMEnvironment() is string _)
+        {
+            try
+            {
+                // introduced in https://github.com/dotnet/msbuild/pull/12581, disables live-updating node display, which wastes tokens
```

Comment:
Nit: The comment is inconsistent with the run command location doing the same.  If you want the patterns to be the same, consider a shared method.

Reply:
I wasn't _quite_ sure what you meant here - but I refactored the run-file sections to operate in terms of the unified `MSBuildArgs` structure and was able to reduce this to one callsite as well.